### PR TITLE
chore(transformers): returns pipelines automatically

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

`load` should return the `pipelines` when `pipeline` is saved with `save`. Before `load` will return `model` and `tokenizer`, and users can load those into pipelines. This could be inconsistent as pipeline tasks will depend on the users to decide what to do with those models. This PR solves this problem:
```python
import transformers
import bentoml

pipeline = transformers.pipeline('text-classification', from_tf=True)
tag = bentoml.transformers.save('distilbert', pipeline)

...
loaded = bentoml.transformers.load(tag, from_tf=True) # type: transformers.pipelines.base.Pipeline
```

## Motivation and Context

Before this PR:
```python
import transformers
import bentoml

pipeline = transformers.pipeline('text-classification', from_tf=True)
tag = bentoml.transformers.save('distilbert', pipeline)

...
model, tokenizer = bentoml.transformers.load(tag)
# the new pipeline might have a different tasks, and is not consistent with our save pipeline.
new_pipeline = transformers.pipeline('text-generation', model=model, tokenizer=tokenizer) 
```
Also now accepts passthrough kwargs for corresponding class with `load`:
```python
config, model, tokenizer = bentoml.transformers.load(tag, return_config=True, model_kwargs={"eos_token_id": "<EOS>"}, tokenizer_kwargs={"use_fast": False}, config_kwargs = {"labels":"..."}
``` 
This is similar to how users can define kwargs for model when loading pipeline, so no learning curve needed. By separating options we can pass those kwargs to the correct loaded module.

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [x] My change reduces project test coverage and requires unit tests to be added
- [x] I have added unit tests covering my code change
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by aarnphm** [comment]
Will wait for #2288 before.

File: bentoml/_internal/frameworks/transformers.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
 from ..utils.pkg import get_pkg_version
 from ..configuration.containers import BentoMLContainer
 
-logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
```
Comment (parano) [inline bentoml/_internal/frameworks/transformers.py:21]: what is this for?

File: bentoml/_internal/frameworks/transformers.py [hunk 2]
```diff
@@ -18,7 +18,8 @@
...
+logger = logging.getLogger("bentoml")
```
Comment (parano) [inline bentoml/_internal/frameworks/transformers.py:22]: let's use `__name__`

File: bentoml/_internal/frameworks/transformers.py (hunk 1 above)
Comment (aarnphm) [reply bentoml/_internal/frameworks/transformers.py:21]: redirect warnings capture from `jax` to logger

**Comment by aarnphm** [comment]
on hold for now for logging discussion

**Comment by aarnphm** [comment]
@bojiang ready now

**Pull request merged**